core/src/sumeragi/network_topology.rs

```diff
@@ -279,6 +300,8 @@ pub enum Role {
     ObservingPeer,
     /// Proxy Tail.
     ProxyTail,
+    /// Undefined. Not part of the topology
+    Undefined,
```

Comment:
Not a hypuge fan of this change. I!d rather this was enclosed in an `Option` in cases where the role is not defined.